sample.yml

```diff
@@ -0,0 +1,26 @@
+- hosts: blx
+  remote_user: root
+  vars:
+          main_iface_name: enp0s25
+          work_process: 2
+          ip_addr: w.x.y.z
+          gateway: w.x.y.z
+
+  tasks:
+        - name: Creating new macvlan interface
+          command: ip link add newiface link {{ main_iface_name  }} type macvlan mode bridge
```

Comment:
Let us not hardcode the interface name:
```
  vars:
    iface: "enp3s0"    # will be provided via Nitin's shell script
    sub_iface: sub0{{ iface }}    # subinterface name adds a prefix to existing interface
```

Set interface state to UP and disable GRO; GSO as well:

```
  tasks:
    - name: Create new sub-interface
      become: yes
      register: sub_iface_result # we should use this in the next play
      shell: 
        cmd: ip link add {{ sub_iface }} link {{ iface  }} type macvlan mode bridge

    - name: Set interface state to up
      become: yes
      shell:
        cmd: ifconfig {{ sub_iface }} up

    - name: Set GRO and GSO to OFF
      become: yes
      shell:
        cmd: sudo ethtool -K {{ sub_iface }} gro off gso off

```

Reply:
KK but anyways Rohit is using another method to change the config file i think he is using variable for the sub-interface name